Bismillahirrahmanirrahim

TUGAS : ERYKKA YUSTARI

#Tugas Individu


1. Tentukan Segment mana dengan profit tertinggi!  [30]
2. Tentukan Category mana dengan sales terbanyak!  [30]
3. Tentukan Sub-Category dengan quantity paling sedikit! [40]





4. **Challange Question!!!  Tentukan Bulan dengan Profit Tertinggi!!!** 

Bonus Nilai 100 di Salah Satu Tugas Lainnya


# Mengimpor Library

In [259]:
import pandas as pd
import numpy as np

#Mengimpor Data Frame dari Format CSV

In [260]:
df = pd.read_csv("https://raw.githubusercontent.com/jokoeliyanto/Kelas-Dasar-Pejuang-Data-2.0/main/Super-Store-Dataset.csv")
df.head()

order_id  order_date customer_id  ...     sales quantity    profit
0  CA-2016-152156   11/8/2016    CG-12520  ...  261.9600        2   41.9136
1  CA-2016-152156   11/8/2016    CG-12520  ...  731.9400        3  219.5820
2  CA-2016-138688   6/12/2016    DV-13045  ...   14.6200        2    6.8714
3  US-2015-108966  10/11/2015    SO-20335  ...  957.5775        5 -383.0310
4  US-2015-108966  10/11/2015    SO-20335  ...   22.3680        2    2.5164

[5 rows x 9 columns]

## Variabel dataframe




Variabel-variabel yang terdapat pada dataset ini adalah sebagai berikut:
1. **order_id** = Nomor ID pembelian(nota)
2. **order_data** = Tanggal pembelian
3. **customer_id** = Nomor ID pembeli
4. **segment** = Segmen produk 
5. **category** = Kategori produk
6. **sub_category** = Sub kategori produk
7. **sales** = Total harga pembelian
8. **quantity** = Jumlah produk dibeli
9. **profit** = Keuntungan

#Menentukan Segmen yang menghasilkan profit tertinggi

In [261]:
df.segment.value_counts()

Consumer       5188
Corporate      3019
Home Office    1781
Name: segment, dtype: int64

In [262]:
df_Consumer=df[df['segment']=='Consumer']
df_Corporate=df[df['segment']=='Corporate']
df_Home_Office=df[df['segment']=='Home Office']

In [263]:
df_Consumer.profit.sum()


128959.22080000001

In [264]:
df_Corporate.profit.sum()

83579.158

In [265]:
df_Home_Office.profit.sum()

58189.7681

Jadi, segmen dengan keuntungan total tertinggi adalah Consumer 

#Menentukan Kategori dengan Sales terbanyak

In [266]:
df.category.value_counts()

Office Supplies    6026
Furniture          2121
Technology         1841
Name: category, dtype: int64

In [267]:
df_OfficeSupplies =df[df['category']=='Office Supplies']
df_Furniture=df[df['category']=='Furniture']
df_Technology=df[df['category']=='Technology']

In [268]:
df_OfficeSupplies.sales.sum()

719047.032

In [269]:
df_Furniture.sales.sum()

741999.7953

In [270]:
df_Technology.sales.sum()

755815.72

Jadi, kategori dengan sales terbanyak adalah Technology

#Menentukan Sub-Category dengan quantity paling sedikit

In [271]:
df.sub_category.value_counts()

Binders        1523
Paper          1370
Furnishings     957
Phones          889
Storage         846
Art             796
Accessories     775
Chairs          617
Appliances      466
Labels          364
Tables          319
Envelopes       254
Bookcases       228
Fasteners       217
Supplies        190
Machines        113
Copiers          64
Name: sub_category, dtype: int64

In [272]:
df_Binders=df[df['sub_category']=='Binders']
df_Paper=df[df['sub_category']=='Paper']
df_Furnishings=df[df['sub_category']=='Furnishings']
df_Phones=df[df['sub_category']=='Phones']
df_Storage=df[df['sub_category']=='Storage']
df_Art=df[df['sub_category']=='Art']
df_Accessories=df[df['sub_category']=='Accessories']
df_Chairs=df[df['sub_category']=='Chairs']
df_Appliances=df[df['sub_category']=='Appliances']
df_Labels=df[df['sub_category']=='Labels']
df_Tables=df[df['sub_category']=='Tables']
df_Envelopes=df[df['sub_category']=='Envelopes']
df_Bookcases =df[df['sub_category']=='Bookcases']
df_Fasteners=df[df['sub_category']=='Fasteners']
df_Supplies=df[df['sub_category']=='Supplies']
df_Machines=df[df['sub_category']=='Machines']
df_Copiers=df[df['sub_category']=='Copiers']

In [273]:
df_Binders.quantity.sum()

5974

In [274]:
df_Paper.quantity.sum()

5178

In [275]:
df_Furnishings.quantity.sum()

3563

In [276]:
df_Phones.quantity.sum()

3289

In [277]:
df_Storage.quantity.sum()

3158

In [278]:
df_Art.quantity.sum()

3000

In [279]:
df_Accessories.quantity.sum()

2976

In [280]:
df_Chairs.quantity.sum()

2356

In [281]:
df_Appliances.quantity.sum()

1729

In [282]:
df_Labels.quantity.sum()


1400

In [283]:
df_Tables.quantity.sum()

1241

In [284]:
df_Envelopes.quantity.sum()

906

In [285]:
df_Bookcases.quantity.sum()

868

In [286]:
df_Fasteners.quantity.sum()

914

In [287]:
df_Supplies.quantity.sum()

647

In [288]:
df_Machines.quantity.sum()

429

In [289]:
df_Machines.quantity.sum()

429

In [290]:
df_Copiers.quantity.sum()

218

Jadi Sub-Category dengan quantity paling sedikit adalah Copiers

#Menentukan Bulan dengan Profit Tertinggi

In [291]:
df['order_date'].min()

'1/1/2017'

In [292]:
df['order_date'].max()

'9/9/2017'

Ternyata format data tanggal ada yang tidak terbaca, maka perlu diperbaiki

In [293]:
df['order_date']=pd.to_datetime(df['order_date'])

In [294]:
#Menambahkan kolom bulan
df['month']=pd.DatetimeIndex(df['order_date']).month
print(df)

            order_id order_date customer_id  ... quantity    profit month
0     CA-2016-152156 2016-11-08    CG-12520  ...        2   41.9136    11
1     CA-2016-152156 2016-11-08    CG-12520  ...        3  219.5820    11
2     CA-2016-138688 2016-06-12    DV-13045  ...        2    6.8714     6
3     US-2015-108966 2015-10-11    SO-20335  ...        5 -383.0310    10
4     US-2015-108966 2015-10-11    SO-20335  ...        2    2.5164    10
...              ...        ...         ...  ...      ...       ...   ...
9983  CA-2014-110422 2014-01-21    TB-21400  ...        3    4.1028     1
9984  CA-2017-121258 2017-02-26    DB-13060  ...        2   15.6332     2
9985  CA-2017-121258 2017-02-26    DB-13060  ...        2   19.3932     2
9986  CA-2017-121258 2017-02-26    DB-13060  ...        4   13.3200     2
9987  CA-2017-119914 2017-05-04    CC-12220  ...        2   72.9480     5

[9988 rows x 10 columns]


In [295]:
df['order_date']=df['order_date'].dt.strftime('%m/%d/%Y')
print(df)

            order_id  order_date customer_id  ... quantity    profit month
0     CA-2016-152156  11/08/2016    CG-12520  ...        2   41.9136    11
1     CA-2016-152156  11/08/2016    CG-12520  ...        3  219.5820    11
2     CA-2016-138688  06/12/2016    DV-13045  ...        2    6.8714     6
3     US-2015-108966  10/11/2015    SO-20335  ...        5 -383.0310    10
4     US-2015-108966  10/11/2015    SO-20335  ...        2    2.5164    10
...              ...         ...         ...  ...      ...       ...   ...
9983  CA-2014-110422  01/21/2014    TB-21400  ...        3    4.1028     1
9984  CA-2017-121258  02/26/2017    DB-13060  ...        2   15.6332     2
9985  CA-2017-121258  02/26/2017    DB-13060  ...        2   19.3932     2
9986  CA-2017-121258  02/26/2017    DB-13060  ...        4   13.3200     2
9987  CA-2017-119914  05/04/2017    CC-12220  ...        2   72.9480     5

[9988 rows x 10 columns]


In [296]:
df.month.value_counts()

11    1469
12    1408
9     1383
10     817
5      735
6      717
7      710
8      706
3      694
4      668
1      381
2      300
Name: month, dtype: int64

In [297]:
order_date=df.groupby('month').sum()
print(order_date)

             sales  quantity      profit
month                                   
1       94924.8356      1475   9134.4461
2       59751.2514      1067  10294.6107
3      168367.0488      2554  23685.7848
4      137762.1286      2447  11587.4363
5      155028.8117      2791  22411.3078
6      152718.6793      2680  21285.7954
7      147238.0970      2705  13832.6648
8      159044.0630      2784  21776.9384
9      307649.9457      5062  36857.4753
10     171623.0667      3095  19464.0765
11     337461.1160      5767  37028.4189
12     325293.5035      5419  43369.1919


Jadi, Bulan dengan profit tertinggi adalah bulan Desember dengan profit sebesar 43369.1919